tests/base/src/test/java/org/wildfly/security/auth/TestLoginModule.java

```diff
@@ -101,42 +102,17 @@ public boolean logout() throws LoginException {
         return true;
     }
 
-    /**
-     * A {@code Group} implementation used in the tests to store the caller principal.
-     */
-    private class TestGroup implements Group {
+    private static class Roles implements Principal {
```

Comment:
@darranl I do not expose this class anywhere, users can create their own principal classes that will be mapped to attribute keys.

Reply:
+1 this is just for our tests.